### PR TITLE
Tests for the Attribute component

### PR DESCRIPTION
**Pull request by EmanueleMinotto**
Unit tests covering the Attribute component.

Also some tests for the Core are included (because used inside the Attribute tests).

This is a split of https://github.com/elcodi/elcodi/pull/991

### CODE REVIEW

File: src/Elcodi/Component/Core/Tests/UnitTest/Entity/AbstractEntityTest.php [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        );
+
+        try {
+            $class->$setter(null);
+            $this->assertNull($class->$getter());
+        } catch (Exception $e) {
+            if ($field['nullable']) {
+                $this->fail("Method $setter of entity $classNamespace is not nullable");
+            }
+        }
```
Comment (EmanueleMinotto) [inline src/Elcodi/Component/Core/Tests/UnitTest/Entity/AbstractEntityTest.php]: imo the try/catch should be wrapped in the `if ($field['nullable']) {` instead

File: src/Elcodi/Component/Core/Tests/UnitTest/Entity/AbstractEntityTest.php [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+            $value,
+            $class->$getter()
+        );
+
+        try {
+            $class->$setter(null);
+            $this->assertNull($class->$getter());
+        } catch (Exception $e) {
+            if ($field['nullable']) {
+                $this->fail("Method $setter of entity $classNamespace is not nullable");
```
Comment (EmanueleMinotto) [inline src/Elcodi/Component/Core/Tests/UnitTest/Entity/AbstractEntityTest.php]: What about a `sprintf('Method %s of entity %s is not nullable', $setter, $classNamespace)` here?

**Comment by mmoreram** [comment]
I'm going to push a change to make all tests green. Can I squash both commits?

**Comment by EmanueleMinotto** [comment]
@mmoreram sure :)

**Pull request merged**